lib/ktl/reassignment_task.rb

```diff
@@ -11,7 +11,12 @@ def initialize(reassigner, plan, shell, options={})
 
     def execute(dryrun = false)
       if @reassigner.reassignment_in_progress?
-        @logger.warn 'reassignment already in progress, exiting'
+        if @reassigner.is_a?(ContinousReassigner)
+          reassignment = @reassigner.load_overflow
+          @reassigner.execute(reassignment)
+        else
+          @logger.warn 'reassignment already in progress, exiting'
+        end
```

Comment:
It would be nicer if this was internal to the reassigners so that this class didn't have to change because you add new reassigner variants.

It feels like this whole if/else block should be possible to rewrite something like:

```ruby
reassignment = @reassigner.load_overflow || @plan.generate
@reassigner.execute(reassignment)
```

and have the different reassigners react differently depending on their capabilities, e.g. warn and do nothing in the case of a non-continous reassigner when a reassigner is in progress, etc.

What I'm saying is that when I see code like

```ruby
if @reassigner.reassignment_in_progress?
  if @reassigner.is_a?(ContinousReassigner)
    reassignment = @reassigner.load_overflow
    @reassigner.execute(reassignment)
  else
```

where every line does something with `@reassigner` and no other objects I think that code belongs inside of that object. This smell is called "feature envy".